src/main/java/com/google/idea/perf/vfstracer/VfsStatTreeTable.kt

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.idea.perf.vfstracer
+
+import com.intellij.ui.components.JBTreeTable
+import com.intellij.ui.treeStructure.treetable.TreeTableModel
+import javax.swing.JTree
+import javax.swing.event.EventListenerList
+import javax.swing.event.TreeModelEvent
+import javax.swing.event.TreeModelListener
+import javax.swing.tree.TreePath
+
+private const val COLUMN_COUNT = 2
+private const val STUB_INDEX_ACCESSES = 0
+private const val PSI_ELEMENT_WRAPS = 1
+
+private operator fun VirtualFileTree.get(index: Int): VirtualFileTree =
+    children.values.toTypedArray()[index]
```

Comment:
I was hesistating on this too. If there was a way to index `TreeMap`s by index, then this could be much cheaper. Another way could be to cache the array conversion with a weak hash map. For now, maybe should be fine because stub indexes are accessed incrementally, but it is something that could be optimized.